src/nvim/message.c

```diff
@@ -618,6 +633,9 @@ void msg_source(int hl_id)
     msg_scroll = true;  // this will take more than one line
     msg(p, hl_id);
     xfree(p);
+    if (is_multihl) {
+      msg_start();  // avoided in msg_keep() but necessary here
```

Comment:
This is a bit questionable since this goes against the point of the `is_multihl` guards, but it currently resolves a failing test related to newline formatting. `msg_start()` emits ext_messages (which we don't want for a single multihl message), while also taking care of formatting the message grid. Once the message grid is removed I want to re-evaluate whether `msg_start/end()` are necessary at all. Currently it results unnecessary message chunk splitting in various places (also seen in https://github.com/neovim/neovim/issues/31901).